seed/seedwriter/writer.go

```diff
@@ -876,6 +880,21 @@ func (w *Writer) checkBase(info *snap.Info, modes []string) error {
 	return w.policy.checkBase(info, modes, w.availableByMode)
 }
 
+func (w *Writer) recordUsageWithThePolicy(modSnaps []*asserts.ModelSnap) {
+	for _, modSnap := range modSnaps {
+		w.policy.recordSnapNameUsage(modSnap.Name)
+	}
+
+	for _, optSnap := range w.optionsSnaps {
+		snapName := optSnap.Name
+		sn := w.localSnaps[optSnap]
+		if sn != nil {
+			snapName = sn.Info.SnapName()
+		}
+		w.policy.recordSnapNameUsage(snapName)
+	}
```

Comment:
this part of the function was a bit unexpected from the usage, shouldn't this be done somewhere else and only once?

Reply:
Ah right - the model step is only once of course